### PR TITLE
fix(dev): upgrading modules including restify-errors

### PR DESCRIPTION
**Pull request by rajatkumar**
## Pre-Submission Checklist

- [ ] Opened an issue discussing these changes before opening the PR
- [ ] Ran the linter and tests via `make prepush`
- [ ] Included comprehensive and convincing tests for changes

## Issues

Closes:

* Issue #
* Issue #
* Issue #

> Summarize the issues that discussed these changes

# Changes

> What does this PR do?
This PR upgrades the dependencies to use the latest versions. This also cleans up a couple of tests, and includes minor changes to and ensures `restify` uses the latest `restify-error` library.
Major libraries updated:
- mocha
- lru-cache
- restify-error
- mime
- vasync
- spdy

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -1164,7 +1165,8 @@ Server.prototype._setupRequest = function _setupRequest(req, res) {
     // we consider a closed request as flushed from metrics point of view
     function onReqAborted() {
         // Request was aborted, override the status code
-        var err = new errors.RequestCloseError();
+        //var err = new errors.RequestCloseError();
```
Comment (misterdjules) [inline lib/server.js]: Why do we want to keep this comment?

File: test/server.test.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ var http = require('http');
 var stream = require('stream');
 
 var errors = require('restify-errors');
-var filed = require('filed');
+// var filed = require('filed');
```
Comment (misterdjules) [inline test/server.test.js]: Why do we want to keep this commented out (as well as the test below)?

File: lib/errorTypes.js [hunk 3]
```diff
@@ -2,5 +2,8 @@
 
 var errors = require('restify-errors');
 
-errors.makeConstructor('RequestCloseError');
-errors.makeConstructor('RouteMissingError');
+// This allows Restify to work with restify-errors v6+
```
Comment (misterdjules) [inline lib/errorTypes.js]: Just out of curiosity: what is the change in restify v6+ that requires this?

File: lib/plugins/accept.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ function acceptParser(accepts) {
             return a;
         })
         .map(function map(a) {
-            return a.indexOf('/') === -1 ? mime.lookup(a) : a;
+            return a.indexOf('/') === -1 ? mime.getType(a) : a;
```
Comment (misterdjules) [inline lib/plugins/accept.js]: What is the difference between `mime.lookup` and `mime.getType`?

File: lib/errorTypes.js (hunk 3 above)
Comment (rajatkumar) [reply lib/errorTypes.js]: In v5 of `restify-errors` the function call to `makeConstructor` added the constructor to `restify-errors` exports. This changed in v6 and the recommendation was that the consumer should keep a reference to the new error constructor.

File: lib/plugins/accept.js (hunk 4 above)
Comment (rajatkumar) [reply lib/plugins/accept.js]: `lookup` was renamed to `getType` in the newer version of `mime`.

File: test/server.test.js (hunk 2 above)
Comment (rajatkumar) [reply test/server.test.js]: I will get rid of it, this is not needed.

File: lib/server.js (hunk 1 above)
Comment (rajatkumar) [reply lib/server.js]: I will get rid of it, this is not needed.

**Pull request merged**